# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/embutils/hotfix-touch-ret.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- embutils-0.19/touch.c.vanilla	2026-08-24 13:46:36.752741476 +0200
+++ embutils-0.19/touch.c	2026-08-24 13:47:29.118777110 +0200
@@ -22,6 +22,8 @@
 
 static const char* errorill="touch: illegal timestamp";
 
+int ret=0;
+
 void oops(const char *fn) {
   int err=errno;
   __write2(errorill);
@@ -41,6 +43,7 @@
   case EIO:		warn("I/O error"); break;
   default: warn("unknown error");
   }
+  ret=1;
 }
 
 void usage() {
@@ -162,4 +165,5 @@
       oops(fn);
     }
   }
+  return ret;
 }
